### ISSUE TITLE
Some suggestions (July '23)

### ISSUE THREAD

**Issue by bennylin** [Ladsgroup/Vandalism-dashboard#14]
Hi, I think your tool is what the community needs to detect vandalism, especially in the short description. While using this, I have some ideas and suggestions:

suggestions:
* Entity ID and Entity title basically the same link, but they're separated. I don't think people need to know the Q-ID itself, and if they click the Entity title, they would know the qid anyway. Or, make one of the link into the item history page, because if it's vandalism, then there's a good chance that the vandal also vandalize other section or other languages in that item.
* Username: instead of linked to profile (user page), please link it to contributions page (just like anons). most of the vandals have empty user page anyway
* Sitelink additions in that language, not just removal. Sitelink changes too.
* Some information about ORES damaging score, how is it counted, and what the color scheme means. I assume red = bad. But the many reds that I saw was good, and some of the vandalism were low on the ORES score (white)
* If you must have the color, you can limit the color only in the ORES column, then you're free to use other color code in other columns, for example in edit summary column: green for addition, yellow for change, red for removal. In user column, red for anon, default (white) for logged in users, and maybe yellow for new users (if possible).
* In the edit summary column, shorten words like "remove sitelink" into "-sitelink", or use a red font "sitelink", green for addition (or +label, +desc), yellow for changes (or ~label, ~desc). And since the [qq]wikipedia will be the majority anyway, just remove any mention of [qq]wiki, an instead, only add "in enwikiversity", "in enwikisource", etc. whenever it's not in wikipedia. (beside [qq]wiki is should've been spelled full 'wikipedia' instead of just 'wiki')

ideas:
* Maybe put a link to quickly revert an obvious vandal (you can put a pre-filled description such as "reverted using wdvd" to promote the tool)
* And on that thought, if you do that, could you also put a marker if that vandalism has been reverted (I remember seeing in page history that there are tag for already-reverted edits)
* Filtered search: search by username, logged in users only, logged out users only, and by qid
* Display the tags of the edit, e.g. Wikidata User Interface, new editor removing a sitelink, etc.
* Separate between Changed label, Added label, and Removed label, Changed desc, Added desc, and Removed desc in the option
* memorize my last setting when I visit https://wdvd.toolforge.org/ (cookie?). For obvious reason, if a user visited the main page for the first time, the language could be set into "en" with maybe 10 limit, so they can see the tool in action immediately.
* count the total contributions of logged in users, and display it next to the username? (or you can colorize users with < 10 edits as red, same as anon, and < 100 edits as yellow, or something)

For the remove sitelink, as per today, when I open https://wdvd.toolforge.org/index.php?lang=de&limit=250&sitelinks=on, I couldn't see my removal which I did earlier today https://www.wikidata.org/w/index.php?title=Q4648&diff=prev&oldid=1910849133, although the last removal in the list was 6 hours after me. Did the tool exclude users from certain group (trusted user, etc)?

Thanks!

**Comment by bennylin** [comment]
Let me know which idea and suggestions you (or other users) supported, and I might try to do some PR with my limited ability.

**Comment by lucaswerkmeister** [comment]
> Entity ID and Entity title basically the same link, but they're separated. I don't think people need to know the Q-ID itself, and if they click the Entity title, they would know the qid anyway. Or, make one of the link into the item history page, because if it's vandalism, then there's a good chance that the vandal also vandalize other section or other languages in that item.

Yeah, making one of the links go to the history sounds reasonable to me (probably the entity ID one, and leave the label as the link to the page itself). Reordering the columns might also make sense, not sure.

> Username: instead of linked to profile (user page), please link it to contributions page (just like anons). most of the vandals have empty user page anyway

I’m not convinced… we don’t just care about vandals, benign edits should also be patrolled after all (see also below). Maybe a “(contribs)” link for registered users like in MediaWiki itself?

> In the edit summary column, shorten words like "remove sitelink" into "-sitelink", or use a red font "sitelink", green for addition (or +label, +desc), yellow for changes (or ~label, ~desc). And since the [qq]wikipedia will be the majority anyway, just remove any mention of [qq]wiki, an instead, only add "in enwikiversity", "in enwikisource", etc. whenever it's not in wikipedia. (beside [qq]wiki is should've been spelled full 'wikipedia' instead of just 'wiki')

Not a fan of this tbh. Using color alone for information (i.e. removing “remove”) is inaccessible. And the names refer to the wiki’s internal database names, so I think changing them would increase confusion for users who are familiar with the db names (and also I don’t like the sound of making the Wikipedias even more special compared to other wikis).

> Maybe put a link to quickly revert an obvious vandal (you can put a pre-filled description such as "reverted using wdvd" to promote the tool)

Don’t think this is doable without a major overhaul, bordering on rewrite, of the tool (you’d need to login via OAuth etc.). The rollback option on the diff page should be close enough IMHO, at least for people with the rollback right.

> memorize my last setting when I visit https://wdvd.toolforge.org/ (cookie?).

Sounds reasonable enough – I guess we currently expect users to bookmark certain links (which also sounds acceptable to me).

> For obvious reason, if a user visited the main page for the first time, the language could be set into "en" with maybe 10 limit, so they can see the tool in action immediately.

Not a fan of special-casing English like that tbh.

> For the remove sitelink, as per today, when I open https://wdvd.toolforge.org/index.php?lang=de&limit=250&sitelinks=on, I couldn't see my removal which I did earlier today https://www.wikidata.org/w/index.php?title=Q4648&diff=prev&oldid=1910849133, although the last removal in the list was 6 hours after me. Did the tool exclude users from certain group (trusted user, etc)?

Yes, the tool only shows *[unpatrolled](https://www.wikidata.org/wiki/Wikidata:Patrol)* edits. You’ve been autoconfirmed [since 2013](https://www.wikidata.org/w/index.php?title=Special:Log&logid=74881379), so your edits get marked as patrolled automatically and don’t need further review. This also gives you the right to mark others’ unpatrolled edits as patrolled, signaling that they don’t need to be reviewed by anyone else anymore.

@Ladsgroup any opinions here?

**Comment by bennylin** [comment]
One more idea: Add a column for the date and time of edit.
